lib/mayaUsd/fileio/chaser/importChaserRegistry.cpp

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright 2021 Apple
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "importChaserRegistry.h"
+
+#include <mayaUsd/base/debugCodes.h>
+#include <mayaUsd/fileio/registryHelper.h>
+
+#include <pxr/base/tf/instantiateSingleton.h>
+#include <pxr/base/tf/refPtr.h>
+#include <pxr/base/tf/registryManager.h>
+#include <pxr/pxr.h>
+
+#include <iterator>
+#include <map>
+#include <string>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+UsdMayaImportChaserRegistry::FactoryContext::FactoryContext(
+    Usd_PrimFlagsPredicate&     returnPredicate,
+    const UsdStagePtr&          stage,
+    const MDagPathArray&        dagPaths,
+    const SdfPathVector&        sdfPaths,
+    const UsdMayaJobImportArgs& jobArgs)
+    : _stage(stage)
+    , _dagPaths(dagPaths)
+    , _sdfPaths(sdfPaths)
+    , _jobArgs(jobArgs)
+{
+}
+
+UsdStagePtr UsdMayaImportChaserRegistry::FactoryContext::GetStage() const { return this->_stage; }
+
+const MDagPathArray& UsdMayaImportChaserRegistry::FactoryContext::GetImportedDagPaths() const
+{
+    return this->_dagPaths;
+}
+
+const SdfPathVector& UsdMayaImportChaserRegistry::FactoryContext::GetImportedPrims() const
+{
+    return this->_sdfPaths;
+}
+
+const UsdMayaJobImportArgs& UsdMayaImportChaserRegistry::FactoryContext::GetImportJobArgs() const
+{
+    return this->_jobArgs;
+}
+
+TF_INSTANTIATE_SINGLETON(UsdMayaImportChaserRegistry);
+
+std::map<std::string, UsdMayaImportChaserRegistry::FactoryFn> _factoryImportRegistry;
+
+UsdMayaImportChaserRegistry::UsdMayaImportChaserRegistry() { }
+UsdMayaImportChaserRegistry::~UsdMayaImportChaserRegistry() { }
```

Comment:
Maybe `= default;` these in the header instead?

Reply:
I realize that just letting the compiler generate default constructors instead is probably easier. Addressed in e8efa0c.